### PR TITLE
Added Noobaa pods to cluster_health_check

### PR DESCRIPTION
**Pull request by AviadP**
Addresses https://github.com/red-hat-storage/ocs-ci/issues/1089

### CODE REVIEW

File: ocs_ci/ocs/cluster.py [hunk 1]
```diff
@@ -334,6 +346,38 @@ def mds_health_check(self, count):
...
+            Noobaa Operator status is not "Running"
+
+        """
+        timeout = 10 * len(self.pods)
+        try:
+            assert self.POD.wait_for_resource(
+                condition='Running', selector=self.noobaa_selector,
+                timeout=timeout, sleep=3,
+            )
+        except AssertionError as e:
```
Comment (ebenahar) [inline ocs_ci/ocs/cluster.py]: I think the try and except here are redundant. It would be ok just to assert

File: ocs_ci/ocs/cluster.py [hunk 2]
```diff
@@ -334,6 +346,38 @@ def mds_health_check(self, count):
...
+            logger.error(e)
+            raise exceptions.NoobaaConditionException(
+                f"Failed to achieve desired Noobaa Operator Status"
+            )
+
+        try:
```
Comment (ebenahar) [inline ocs_ci/ocs/cluster.py]: same

File: ocs_ci/ocs/resources/pod.py [hunk 3]
```diff
@@ -113,7 +112,7 @@ def get_fio_results(self):
         try:
             if self.fio_thread.running():
                 for sample in TimeoutSampler(
-                    timeout=FIO_TIMEOUT, sleep=3, func=self.fio_thread.done
+                        timeout=FIO_TIMEOUT, sleep=3, func=self.fio_thread.done
```
Comment (ebenahar) [inline ocs_ci/ocs/resources/pod.py]: delete extra indentation

File: ocs_ci/ocs/resources/pod.py [hunk 4]
```diff
@@ -128,7 +127,7 @@ def get_fio_results(self):
             raise
 
     def exec_cmd_on_pod(
-        self, command, out_yaml_format=True, secrets=None, timeout=600, **kwargs
+            self, command, out_yaml_format=True, secrets=None, timeout=600, **kwargs
```
Comment (ebenahar) [inline ocs_ci/ocs/resources/pod.py]: delete extra indentation

File: ocs_ci/ocs/resources/pod.py [hunk 5]
```diff
@@ -246,8 +245,8 @@ def workload_setup(self, storage_type, jobs=1):
         self.wl_setup_done = True
 
     def run_io(
-        self, storage_type, size, io_direction='rw', rw_ratio=75,
-        jobs=1, runtime=60, depth=4, fio_filename=None
+            self, storage_type, size, io_direction='rw', rw_ratio=75,
```
Comment (ebenahar) [inline ocs_ci/ocs/resources/pod.py]: delete extra indentation

File: ocs_ci/ocs/resources/pod.py [hunk 6]
```diff
@@ -409,7 +408,7 @@ def get_ceph_tools_pod():
     running_ct_pods = list()
     for pod in ct_pod_items:
         if ocp_pod_obj.get_resource_status(
-            pod.get('metadata').get('name')
+                pod.get('metadata').get('name')
```
Comment (ebenahar) [inline ocs_ci/ocs/resources/pod.py]: same

File: ocs_ci/ocs/resources/pod.py [hunk 7]
```diff
@@ -753,8 +752,8 @@ def get_pod_count(label, namespace=None):
 
 
 def get_cephfsplugin_provisioner_pods(
-    cephfsplugin_provisioner_label=constants.CSI_CEPHFSPLUGIN_PROVISIONER_LABEL,
-    namespace=None
+        cephfsplugin_provisioner_label=constants.CSI_CEPHFSPLUGIN_PROVISIONER_LABEL,
```
Comment (ebenahar) [inline ocs_ci/ocs/resources/pod.py]: same

File: ocs_ci/ocs/resources/pod.py [hunk 8]
```diff
@@ -776,8 +775,8 @@ def get_cephfsplugin_provisioner_pods(
 
 
 def get_rbdfsplugin_provisioner_pods(
-    rbdplugin_provisioner_label=constants.CSI_RBDPLUGIN_PROVISIONER_LABEL,
-    namespace=None
+        rbdplugin_provisioner_label=constants.CSI_RBDPLUGIN_PROVISIONER_LABEL,
```
Comment (ebenahar) [inline ocs_ci/ocs/resources/pod.py]: same

File: ocs_ci/ocs/resources/pod.py [hunk 9]
```diff
@@ -1017,7 +1016,7 @@ def plugin_provisioner_leader(interface, namespace=None):
                 # Ensure that there is no non leader message logged after
                 # the last occurrence of leader message
                 if not any(
-                    non_leader_msg in msg for msg in log_list[:curr_index]
+                        non_leader_msg in msg for msg in log_list[:curr_index]
```
Comment (ebenahar) [inline ocs_ci/ocs/resources/pod.py]: same

File: ocs_ci/ocs/cluster.py [hunk 10]
```diff
@@ -234,6 +240,12 @@ def cluster_health_check(self, timeout=None):
         except exceptions.MDSCountException as e:
             logger.error(e)
             raise exceptions.CephHealthException("Cluster health is NOT OK")
+
+        try:
+            self.noobaa_health_check()
+        except exceptions.NoobaaConditionException as e:
```
Comment (ebenahar) [inline ocs_ci/ocs/cluster.py]: You are not raising this exception from noobaa_health_check() anymore

**Comment by AviadP** [comment]
verified locally

**Pull request merged**